src/Goteo/Model/Traits/ImpactDataRelationsTrait.php

```diff
@@ -48,6 +48,18 @@ public function addImpactData(ImpactData $impact_data, int $order) {
         return $this;
     }
 
+    public function hasImpactData(ImpactData $impactData): bool {
+        $tb = strtolower($this->getTable());
+        $rel = $this->getImpactDataTable();
+
+        $sql = "SELECT impact_data.*
+            FROM `$rel`
+            INNER JOIN impact_data ON impact_data.id = `$rel`.impact_data_id
+            WHERE `$rel`.impact_data_id = ?";
```

Comment:
The SQL could be optimized by putting selecting only the id in the SELECT. Since it could return a lot of information which is not relevant to know if there is related `ImpactData`.

Example:
```
SELECT impact_data.id
FROM `$rel`
INNER JOIN impact_data ON impact_data.id = `$rel`.impact_data_id
WHERE `$rel`.impact_data_id = ?
```

Reply:
True! In fact, we don't even need the join with impact data, since we only want to check that it exists in the relation table.